dql/cli.py

```diff
@@ -198,21 +204,37 @@ def initialize(
         self.throttle = TableLimits()
         self.throttle.load(self.conf["_throttle"])
 
+    def with_exception_handling(self, func_name, *args, **kwargs):
+        """ Will run the given function with the supplied args and kwargs and handle any exceptions. """
+        try:
+            func = getattr(self, func_name)
+            # console.log("Running a function: " + func_name + "() with *args, **kwargs", log_locals=True)
+            func(*args, **kwargs)
+        except KeyboardInterrupt:
+            console.print("KeyboardInterrupt by user")
+        except botocore.exceptions.BotoCoreError as e:
+            console.log("BotoCoreError: ", e)
+        except ParseException as e:
+            console.log("Engine: ParseException")
+            syntax = Syntax(
+                self.engine.pformat_exc(e),
+                "sql",
+                theme="monokai",
+                line_numbers=True,
+                word_wrap=True,
+            )
+            console.print(Panel(syntax, title="Engine Details", expand=False))
+        except SyntaxError as e:
+            console.log(e)
+        except Exception:
+            console.print_exception()
+
```

Comment:
You could do this as a context manager and avoid having to do a `getattr` for the method. Slightly nicer because tools like pylint and mypy (if we ever add Python 3 typing in the future) will be able to follow the logic.

```python
from contextlib import contextmanager

@contextmanager
def catch_and_print_errors(engine: Engine):
  try:
    yield
  except:
    ...
```

And then use it like so:
```python
with catch_and_print_errors(self.engine):
  self.cmdloop()
```

Reply:
I have to learn about this context manager.

Reply:
ok, will do.